autoload/neomake.vim

```diff
@@ -512,7 +512,7 @@ function! s:AddExprCallback(maker) abort
             call setqflist(list, 'r')
         endif
     endif
-    if counts_changed
+    if counts_changed && has_key(a:maker, 'bufnr')
         call s:neomake_hook('NeomakeCountsChanged', {
                     \ 'file_mode': a:maker.file_mode,
                     \ 'bufnr': a:maker.bufnr})
```

Comment:
Here were my steps: 

1. Open neovim on a folder that has a "project" inside of it.  For example, this could be a rust project, or a typescript project (if you merge this change).
2. Without opening any files, run `NeomakeProject cargo` (for rust) or `Neomake typescript_project` for typescript.
3. Neomake crashes